resources/testharness.js

```diff
@@ -483,7 +483,10 @@ policies and contribution forms [3].
     }
 
     function is_service_worker(worker) {
-        return 'ServiceWorker' in self && worker instanceof ServiceWorker;
+        // The worker object may be from another execution context,
+        // so do not use instanceof here.
+        return 'ServiceWorker' in self &&
+            Object.prototype.toString.call(worker) == '[object ServiceWorker]';
```

Comment:
I spoke with @bzbarsky a bit and he suggested something like this would be better:

```
try {
  Object.getOwnPropertyDescriptor(self.ServiceWorker.prototype, "scriptURL").get.call(worker);
  return true;
} catch (e) {
  return false;
}
```

This uses the webidl requirement that interface methods can only be called on objects of the correct instance type.  Boris believes this should work across compartments/contexts.

This is better than the proposed change because code cannot lie about being a ServiceWorker object by changing their @@toStringTag.